CMakeLists.txt

```diff
@@ -0,0 +1,347 @@
+option(BUILD_BENCHMARK "builds benchmark project" ON)
+
+option(BUILD_UTILITY_GEOMETRY_TO_WKB "builds the utility program geometry_to_wkb" ON)
+option(BUILD_UTILITY_MAPNIK_INDEX "builds the utility program mapnik_index" ON)
+option(BUILD_UTILITY_MAPNIK_RENDER "builds the utility program mapnik_render" ON)
+option(BUILD_UTILITY_OGRINDEX "builds the utility program ogrindex" OFF)
+option(BUILD_UTILITY_PGSQL2SQLITE "builds the utility program pgsql2sqlite" ON)
+option(BUILD_UTILITY_SHAPEINDEX "builds the utility program shapeindex" ON)
+option(BUILD_UTILITY_SVG2PNG "builds the utility program svg2png" ON)
+
+option(USE_BOOST_REGEX_ICU_WORKAROUND "if you don't use your system libraries and get double linked icu libraries set this to ON" OFF)
+
+set(CMAKE_CXX_STANDARD 14 CACHE STRING "Sets the c++ standard. c++14 is minimum.")
+message(STATUS "Using c++${CMAKE_CXX_STANDARD}")
+# https://cmake.org/cmake/help/latest/prop_tgt/CXX_EXTENSIONS.html
+set(CXX_EXTENSIONS OFF CACHE STRING "Enables the compiler specific extensions.") # Fallsback to -std=c++<ver> if off
+message(STATUS "Using c++ extensions: ${CXX_EXTENSIONS}")
+# https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD_REQUIRED.html#prop_tgt:CXX_STANDARD_REQUIRED
+set(CXX_STANDARD_REQUIRED ON) # require the specified CMAKE_CXX_STANDARD
+
+if(WIN32)
+    set(DEFAULT_BIN_DIR bin)
+    set(DEFAULT_LIB_DIR lib)
+    set(DEFAULT_ARCHIVE_DIR lib)
+else()
+    include(GNUInstallDirs)
+    set(DEFAULT_BIN_DIR ${CMAKE_INSTALL_BINDIR})
+    set(DEFAULT_LIB_DIR ${CMAKE_INSTALL_LIBDIR})
+    set(DEFAULT_ARCHIVE_DIR ${CMAKE_INSTALL_LIBDIR})
+endif()
+# https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#runtime-output-artifacts
+set(MAPNIK_BIN_DIR ${DEFAULT_BIN_DIR} CACHE STRING "Install directory for binaries")
+# https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#library-output-artifacts
+set(MAPNIK_LIB_DIR ${DEFAULT_LIB_DIR} CACHE STRING "Install directory for libraries")
+# https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#archive-output-artifacts
+set(MAPNIK_ARCHIVE_DIR ${DEFAULT_ARCHIVE_DIR} CACHE STRING "Install directory for archives")
+
+set(INSTALL_CMAKE_DIR ${MAPNIK_LIB_DIR}/cmake/mapnik CACHE STRING "Install directory of the cmake targets")
+
+if(WIN32)
+    set(DEFAULT_PLUGINS_INSTALL_DIR ${MAPNIK_BIN_DIR}/mapnik/input)
+else()
+    set(DEFAULT_PLUGINS_INSTALL_DIR ${MAPNIK_LIB_DIR}/mapnik/input)
+endif()
+set(PLUGINS_INSTALL_DIR ${DEFAULT_PLUGINS_INSTALL_DIR} CACHE STRING "installs the plugins in the specified directory")
+message(STATUS "Installing plugins to ${PLUGINS_INSTALL_DIR}")
+
+set(FONTS_INSTALL_DIR ${MAPNIK_BIN_DIR}/fonts CACHE STRING "installs the fonts in the specified directory")
+message(STATUS "Installing fonts to ${FONTS_INSTALL_DIR}")
+
+
+# Begin project configuration
+mapnik_find_threads()
+mapnik_find_package(ICU REQUIRED COMPONENTS uc i18n data)
+mapnik_find_package(Boost 1.74 REQUIRED COMPONENTS filesystem system regex)
```

Comment:
Should this be `1.61` ? https://github.com/mapnik/mapnik/blob/master/SConstruct#L67

Building on older systems (RHEL 7 in my case) makes it hard if boost is too new